homeassistant/components/raincloud.py

```diff
@@ -0,0 +1,181 @@
+    'battery': '%',
+    'is_watering': '',
+    'manual_watering': '',
+    'next_cycle': '',
+    'rain_delay': 'days',
+    'status': '',
+    'watering_time': 'min',
+}
+
+BINARY_SENSORS = ['is_watering', 'status']
+
+SENSORS = ['battery', 'next_cycle', 'rain_delay', 'watering_time']
+
+SWITCHES = ['auto_watering', 'manual_watering']
+
+SCAN_INTERVAL = timedelta(seconds=20)
+
+SIGNAL_UPDATE_RAINCLOUD = "raincloud_update"
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+            cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Melnor RainCloud component."""
+    conf = config[DOMAIN]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+    scan_interval = conf.get(CONF_SCAN_INTERVAL)
+
+    try:
+        from raincloudy.core import RainCloudy
+
+        raincloud = RainCloudy(username=username, password=password)
+        if not raincloud.is_connected:
+            raise HTTPError
+        hass.data[DATA_RAINCLOUD] = RainCloudHub(raincloud)
+    except (ConnectTimeout, HTTPError) as ex:
+        _LOGGER.error("Unable to connect to Rain Cloud service: %s", str(ex))
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+        return False
+
+    def hub_refresh(event_time):
+        """Call Raincloud hub to refresh information."""
+        _LOGGER.debug("Updating RainCloud Hub component.")
+        raincloud = hass.data[DATA_RAINCLOUD]
+        raincloud.data.update()
+
+        dispatcher_send(hass, SIGNAL_UPDATE_RAINCLOUD)
+
+    # Call the Raincloud API to refresh updates
+    track_time_interval(hass, hub_refresh, scan_interval)
+
+    return True
+
+
+class RainCloudHub(object):
+    """Base class for all Raincloud entities."""
+
+    def __init__(self, data):
+        """Initialize the entity."""
+        self.data = data
+
+
+class RainCloudEntity(Entity):
+    """Entity class for RainCloud devices."""
+
+    def __init__(self, data, sensor_type):
+        """Initialize the entity."""
+        self.data = data
+        self._sensor_type = sensor_type
+        self._name = "{0} {1}".format(
+            self.data.name, KEY_MAP.get(self._sensor_type))
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        async_dispatcher_connect(
+            self.hass, SIGNAL_UPDATE_RAINCLOUD, self._update_callback)
+
+    def _update_callback(self):
+        """Callback update method."""
+        self.update()
```

Comment:
I think you should use `self.schedule_update_ha_state(True)`

Reply:
Thanks!